homeassistant/components/remote/harmony.py

```diff
@@ -0,0 +1,163 @@
+https://home-assistant.io/components/remote.harmony/
+
+"""
+
+import logging
+from homeassistant.const import (
+    CONF_NAME, CONF_HOST, CONF_PORT)
+from homeassistant.components.remote import PLATFORM_SCHEMA
+from homeassistant.util import slugify
+import homeassistant.components.remote as remote
+import homeassistant.helpers.config_validation as cv
+import voluptuous as vol
+
+REQUIREMENTS = ['pyharmony==1.0.12']
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_DEVICE = 'device'
+ATTR_COMMAND = 'command'
+ATTR_ACTIVITY = 'activity'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_NAME): cv.string,
+    vol.Required(CONF_HOST): cv.string,
+    vol.Required(CONF_PORT): cv.string,
+    vol.Required(ATTR_ACTIVITY, default='Not_Set'): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup Harmony platform."""
+    import pyharmony
+    import urllib.parse
+    name = config.get(CONF_NAME)
+    host = config.get(CONF_HOST)
+    port = config.get(CONF_PORT)
+    _LOGGER.info('Loading Harmony platform: ' + name)
+
+    harmony_conf_file = hass.config.path('harmony_' + slugify(name) + '.conf')
+
+    try:
+        _LOGGER.debug('calling pyharmony.ha_get_token for remote at: ' +
+                      host + ':' + port)
+        token = urllib.parse.quote_plus(pyharmony.ha_get_token(host, port))
+    except ValueError as err:
+        _LOGGER.critical(err.args[0] + ' for remote: ' + name)
+        return False
+
+    _LOGGER.debug('received token: ' + token)
+    add_devices([HarmonyRemote(config.get(CONF_NAME),
+                               config.get(CONF_HOST),
+                               config.get(CONF_PORT),
+                               config.get(ATTR_ACTIVITY),
+                               harmony_conf_file,
+                               token)])
+    return True
+
+
+class HarmonyRemote(remote.RemoteDevice):
+    """Remote representation used to control a Harmony device."""
+
+    def __init__(self, name, host, port, activity, path, token):
+        """Initialize HarmonyRemote class."""
+
+        import pyharmony
+        from pathlib import Path
+
+        _LOGGER.debug('HarmonyRemote device init started for: ' + name)
+        self._name = name
+        self._ip = host
+        self._port = port
+        self._state = None
+        self._current_activity = None
+        self._default_activity = activity
+        self._token = token
+        self._config_path = path
+        _LOGGER.debug('retrieving harmony config using token: ' + token)
+        self._config = pyharmony.ha_get_config(self.token, host, port)
+        if not Path(self._config_path).is_file():
+            _LOGGER.debug('writing harmony configuration to file: ' + path)
+            pyharmony.ha_write_config_file(self._config, self._config_path)
+
+    @property
+    def name(self):
+        """Return the Harmony device's name."""
+        return self._name
+
+    @property
+    def token(self):

```

Comment:
Only make things properties if they are going to be consumed by the component. That is not the case with this property so you can just refer to it using `self._token` in your code and remove this property.